developers/ide/vscode.md

```diff
@@ -5,7 +5,87 @@ title: Visual Studio Code
 
 # VSCode
 
-::: tip TODO
-We are currently reworking how to setup a perfect development enviroment.
-A new step by step guide will appear here soon.
-:::
+The following shows the steps for a Win10 system but should be similar for OSX/Linux.  Please note that you must double up on backward slashes on a windows system (`c:\\something`).  On linux/osx, simply use a forward slash (`\etc\something`) .
+
+## Pre-requesites
+
+1. Must have Visual Studio Code (VSCode) installed
+2. Java 8+ SDK (not the JRE) installed and either JAVA_HOME or JDK_HOME pointing to it
+3. Base development directory (example: `c:\users\me\development`).  The `%BASE%` references below will be to this directory.
+4. Maven installed and the `mvn` command is in your path (open a console and type `mvn -version` to verify)
+5. openHAB 2.5+ is already installed
+
+## References
+
+1. [Addon development thread](https://community.openhab.org/t/addon-development/73935/19)
+2. [VSCode with Java](https://code.visualstudio.com/docs/languages/java)
+3. [VSCode with Java debugging](https://code.visualstudio.com/docs/java/java-debugging)
+4. [openHAB Migration to BND](https://github.com/maggu2810/openhab-dev-doc/blob/master/maven-bnd.md)
+
+## Steps
+
+1. Install Java Extension Pack for VSCode (https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack)
+   
+2. Clone the addons (https://github.com/openhab/openhab2-addons.git or your own fork) to %BASE%\openhab2-addons
+   
+3. Ensure the bundle builds correctly (natively with maven)
+   1. Open console to `%BASE%\openhab2-addons\bundles\org.openhab.binding.sony` (or pick any other bundle)
+   2. `mvn clean install -DskipChecks` in the console to build the bundle
+   3. Should produce a jar file in the 'target' directory (example: `%BASE%\openhab2-addons\bundles\org.openhab.binding.sony\target\org.openhab.binding.sony-2.5.0-SNAPSHOT.jar`)
+   
+4. Open VSCode and then open the folder of the bundle (example: `%BASE%\openhab2-addons\bundles\org.openhab.binding.sony`)
```

Comment:
Forward slashes should work on newer windows systems as well, don't they?. Maybe use forward slashes everywhere if that is the case?

Reply:
Yes - but seeing (and really - writing) something like "c://openhab25" is just feels too weird and unnatural if you've been using windows for awhile.  No problem to do it - just feels weird for anyone that's used windows for any length of time (and if they naturally use backward slashes - they still need a note to double up on them).  How about I put in the note section about this - something to the effect (not exact wording) - "Use either forward or backward slashes on windows environment but if you use backward slashes - you need to double up on them".  Then I'll switch everything to forward slashes

Reply:
Yes that should be in the notes. It's just very cool and helpful that we can nowadays use the same path separater on all operating systems.